### PR TITLE
commit 02.03.2021,12:33

### PR DESCRIPTION
**Pull request by Mihran2996**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -29,14 +29,37 @@
...
+        <dependency>
+            <groupId>org.modelmapper.extensions</groupId>
+            <artifactId>modelmapper-spring</artifactId>
+            <version>2.3.0</version>
+        </dependency>
+        <dependency>
+            <groupId>commons-io</groupId>
+            <artifactId>commons-io</artifactId>
+            <version>2.7</version>
+        </dependency>
```
Comment (Arto94) [inline pom.xml:50]: arajarkum em dependency-neri versianer@ texadrel verevum maven property tagi mej.

File: src/main/java/onlineshop/demo/config/WebSecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    public void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.userDetailsService(userDetailsService)
+                .passwordEncoder(passwordEncoder);
+    }
+
+    @Bean
+    public PasswordEncoder encoder() {
+
+        return new BCryptPasswordEncoder();
+    }
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/config/WebSecurityConfig.java:61]: arajarkum em heracnel avelord bacatner@ methodi { bacveluc heto miangamic return

File: src/main/java/onlineshop/demo/endpoint/AddressEndpoint.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/rest/address")
+public class AddressEndpoint {
+    @Autowired
+    private AddressServiceImpl addressService;
+    @Autowired
+    private UserServiceImpl userService;
+
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/endpoint/AddressEndpoint.java:23]: arajarkum em injectner chanel fielderi makardakov, dra poxaren sarkenq constructor u constructri mijocov autowired anenq

File: src/main/java/onlineshop/demo/endpoint/AddressEndpoint.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/rest/address")
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/endpoint/AddressEndpoint.java:17]: partadir chi nshel /rest , karox enq uxxaki /adrress

File: src/main/java/onlineshop/demo/endpoint/AddressEndpoint.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    @Autowired
+    private AddressServiceImpl addressService;
+    @Autowired
+    private UserServiceImpl userService;
+
+    @PostMapping("/create")
+    public ResponseEntity create(@RequestBody Address address, @AuthenticationPrincipal CurrentUser principal) {
+        User user = principal.getUser();
+        address.setUser(user);
+        user.setAddresses2(addressService.addAddress(user.getAddresses(), address));
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/endpoint/AddressEndpoint.java:28]: qani vor verevum arden haytarael en /address apa ays depqum karox enq url datark toxnel
qani vor PostMapping - arden haskanum enq vor create e linelu

File: src/main/java/onlineshop/demo/endpoint/CardEndpoint.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+public class CardEndpoint {
+    @Autowired
+    private CardServiceImpl cardService;
+
+    @PutMapping("/update/{flowerId},{count}")
+    public Card create(@PathVariable(value = "flowerId", required = false) int flowerId, @PathVariable(value = "count", required = false) int count,
+                       @AuthenticationPrincipal CurrentUser user) {
+        return cardService.updateCart(flowerId, count, user.getUser());
+    }
+
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/endpoint/CardEndpoint.java:21]: npanapes,nayel verevum
arajarkum em methodi @ndunox argumentner@ irar tak texadrenq , ete dranq kam shat erkar en petq e scroll anenq kam 3 paramteric aveli depqum

File: src/main/java/onlineshop/demo/endpoint/FlowerEndpoint.java [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    @PostMapping("/create")
+    public ResponseEntity create(@RequestBody Flower flower,MultipartFile file ) throws IOException {
+        return ResponseEntity.ok(flowerService.addFlower(flower,file));
+    }
+
+    @PutMapping("/put")
+    public ResponseEntity update(@RequestBody Flower flower) {
+        return ResponseEntity.ok(flowerService.save(flower));
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/endpoint/FlowerEndpoint.java:37]: nmanapes kareli poxarinel , nael verevum

File: src/main/java/onlineshop/demo/endpoint/FlowerEndpoint.java [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+    @GetMapping("/flowers")
+    public List<Flower> getFlowers() {
+        return flowerService.getFlowers();
+    }
+
+    @PostMapping("/create")
+    public ResponseEntity create(@RequestBody Flower flower,MultipartFile file ) throws IOException {
+        return ResponseEntity.ok(flowerService.addFlower(flower,file));
+    }
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/endpoint/FlowerEndpoint.java:33]: Flower flower,MultipartFile file toxeri dasavorutyan xndir ka arajarkum em bolor texerum ctrl alt l formatavorum

File: src/main/java/onlineshop/demo/endpoint/FlowerEndpoint.java [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+        return ResponseEntity.ok(flowerService.save(flower));
+    }
+
+    //jnjumeq caxik@ @st id-i
+    @DeleteMapping("/delete/{id}")
+    public ResponseEntity delete(@PathVariable int id) {
+        flowerService.delete(id);
+        return ResponseEntity.ok().build();
+    }
+
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/endpoint/FlowerEndpoint.java:46]: arjarkum em commentner@ grel angleren  isk ays depqum commenti poxaren ogtagorcel jav doc nerbernelov java doc plugin@ idea-ic

File: src/main/java/onlineshop/demo/endpoint/UserEndpoint.java [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+    public List<User> users() {
+        return userService.getAllUsers();
+    }
+
+    //stexcum enq user
+    @PostMapping("/create")
+    public ResponseEntity create(@RequestBody UserDto user, Locale locale, MultipartFile file) throws IOException, MessagingException {
+        return ResponseEntity.ok(userService.addUser(user, locale,file));
+    }
+
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/endpoint/UserEndpoint.java:62]: arden grel commentneri veraberyal

File: src/main/java/onlineshop/demo/endpoint/UserEndpoint.java [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+    @GetMapping("/activate/{email},{token}")
+    public ResponseEntity activatedAccount(@PathVariable("email") String email,
+                                           @PathVariable("token") String token) {
+        Optional<User> userFromDb = userService.getByEmail(email);
+        if (userFromDb.isPresent()) {
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/endpoint/UserEndpoint.java:67]: arajarkum em stexcel dto class u mej@ erku dashter@ tal u @ndunel vorpes RequestBody, arajarkum em poxarinel nman kerp  kodi bolor hnaravor texerum

File: src/main/java/onlineshop/demo/repository/FlowerRepository.java [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+
+    @Query(value = "select * from flower where category=:category and price>=100 and price<=:price", nativeQuery = true)
+    List<Flower> filterByCategoryAndPriceDiapason_100_From_200(@Param("category") String category, @Param("price") double price);
+
+    @Query(value = "select * from flower where category=:category and price>=:price", nativeQuery = true)
+    List<Flower> filterByCategoryAndPriceMin_200(@Param("category") String category, @Param("price") double price);
+
+    @Query(value = "select * from flower where  price<=:price", nativeQuery = true)
+    List<Flower> filterByPriceMax_50(@Param("price") double price);
+
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/repository/FlowerRepository.java:28]: javayum chunenq methodner filterByPriceMax_50  _-simvol@ chenq ogtagorcum . ays depqum uxxaki
filterByPrice50

arajarkum em price bolor methodner@ miavorel grel mek custom query

File: src/main/java/onlineshop/demo/service/UserServiceImpl.java [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+    public List<User> getAllUsers() {
+        return userRepository.findAll();
+    }
+
+    @Override
+    public User addUser(UserDto userDto, Locale locale, MultipartFile file) throws IOException, MessagingException {
+        String name = System.currentTimeMillis() + "_" + file.getOriginalFilename();
+        File image = new File(uploadDir, name);
+        file.transferTo(image);
+        userDto.setProfilePic(name);
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/service/UserServiceImpl.java:48]: arajarkum em sarqel FileUtil static class u mej@ static methodner String name = System.currentTimeMillis() + "_" + file.getOriginalFilename(); orinaki hamar. u ete kodum kan krknvox hatvacner kareli e miavorel mi texum utility classum

File: src/main/java/onlineshop/demo/service/UserServiceImpl.java [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+    @Autowired
+    private EmailService mailService;
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private ModelMapper modelMapper;
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Override
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/service/UserServiceImpl.java:38]: autowired masin nshel em verevum

File: src/main/java/onlineshop/demo/service/FlowerServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+    @Override
+    public Flower addFlower(Flower flower,MultipartFile file) throws IOException {
+        String name = System.currentTimeMillis() + "_" + file.getOriginalFilename();
+        File image = new File(uploadDir, name);
+        file.transferTo(image);
+        final List<Image> images = flower.getImages();
+        for (Image image1 : images) {
+            image1.setName(name);
+        }
+        flower.setImages(images);
```
Comment (Arto94) [inline src/main/java/onlineshop/demo/service/FlowerServiceImpl.java:37]: arajarkum em xusapenq popxakanneri anunic heto inch vor tver texadreluc

**Comment by Mihran2996** [comment]
nshvac terutyunner@ uxxvace

File: src/main/java/onlineshop/demo/model/Gender.java [hunk 16]
```diff
@@ -0,0 +1,5 @@
+package onlineshop.demo.model;
+
+public enum Gender {
+    MALE,FEMALE
```
Comment (Mihran2996) [inline src/main/java/onlineshop/demo/model/Gender.java:4]: for test

**Comment by Mihran2996** [comment]
project finished

**Pull request closed without merging**